### PR TITLE
[5.7] Remove elses

### PR DESCRIPTION
**Pull request by omarkdev**
Its removed some `else's` that are considered useless in the code, to improve code reading and to decrease the cyclomatic complexity

### CODE REVIEW

File: src/Illuminate/Container/Container.php [hunk 1]
```diff
@@ -649,11 +649,9 @@ protected function resolve($abstract, $parameters = [])
         // We're ready to instantiate an instance of the concrete type registered for
         // the binding. This will instantiate the types, as well as resolve any of
         // its "nested" dependencies recursively until all have gotten resolved.
-        if ($this->isBuildable($concrete, $abstract)) {
-            $object = $this->build($concrete);
-        } else {
-            $object = $this->make($concrete);
-        }
+        $object = $this->isBuildable($concrete, $abstract)
```
Comment (jmarcher) [inline src/Illuminate/Container/Container.php]: How is this any better than the IFs? I find `if(){ ... }else{ ... }` much clearer.

File: src/Illuminate/Database/Eloquent/ModelNotFoundException.php [hunk 2]
```diff
@@ -35,11 +35,9 @@ public function setModel($model, $ids = [])
 
         $this->message = "No query results for model [{$model}]";
 
-        if (count($this->ids) > 0) {
-            $this->message .= ' '.implode(', ', $this->ids);
-        } else {
-            $this->message .= '.';
-        }
+        $this->message = count($this->ids) > 0
```
Comment (jmarcher) [inline src/Illuminate/Database/Eloquent/ModelNotFoundException.php]: How is this any better than the IFs? I find `if(){ ... }else{ ... }` much clearer.

File: src/Illuminate/Foundation/Console/ObserverMakeCommand.php [hunk 3]
```diff
@@ -69,11 +69,9 @@ protected function replaceModel($stub, $model)
 
         $namespaceModel = $this->laravel->getNamespace().$model;
 
-        if (Str::startsWith($model, '\\')) {
-            $stub = str_replace('NamespacedDummyModel', trim($model, '\\'), $stub);
-        } else {
-            $stub = str_replace('NamespacedDummyModel', $namespaceModel, $stub);
-        }
+        $stub = Str::startsWith($model, '\\')
```
Comment (jmarcher) [inline src/Illuminate/Foundation/Console/ObserverMakeCommand.php]: How is this any better than the IFs? I find `if(){ ... }else{ ... }` much clearer.

**Comment by sisve** [comment]
> ... considered useless in the code, to improve code reading ...

**[Citation needed]**

Do people really find the `?:` operator easier to read than the usual if-clauses? How about nested `?:` operators and php's unique precedence rules?

> decrease the cyclomatic complexity

I don't think any of these changes reduces cyclomatic complexity. All the conditionals are still there, you have just moved them around a bit.

File: src/Illuminate/Support/Arr.php [hunk 4]
```diff
@@ -256,11 +256,11 @@ public static function forget(&$array, $keys)
             while (count($parts) > 1) {
                 $part = array_shift($parts);
 
-                if (isset($array[$part]) && is_array($array[$part])) {
-                    $array = &$array[$part];
-                } else {
+                if (! isset($array[$part]) || ! is_array($array[$part])) {
```
Comment (jmarcher) [inline src/Illuminate/Support/Arr.php]: This is harder to read `if true and true` is in my humble opinion better than `if not false or not false`

File: src/Illuminate/Validation/ValidationRuleParser.php [hunk 5]
```diff
@@ -195,11 +195,9 @@ public static function parse($rules)
             return [$rules, []];
         }
 
-        if (is_array($rules)) {
-            $rules = static::parseArrayRule($rules);
-        } else {
-            $rules = static::parseStringRule($rules);
-        }
+        $rules = is_array($rules)
```
Comment (jmarcher) [inline src/Illuminate/Validation/ValidationRuleParser.php]: How is this any better than the IFs? I find `if(){ ... }else{ ... }` much clearer.

File: src/Illuminate/Database/Eloquent/ModelNotFoundException.php (hunk 2 above)
Comment (phannaly) [reply src/Illuminate/Database/Eloquent/ModelNotFoundException.php]: This shortens make a bit confuse. Teneray not always good.

**Comment by mfn** [comment]
> Do people really find the `?:` operator easier to read than the usual if-clauses?

I do, for the easy cases, yes.

To me, a regular if spans 5 lines of verbosity:
```php
if (foo) {
  $baz = doFoo();
} else {
  $baz = doBar();
}
```
I prefer the short version as long as the whole if is trivial:
```php
$baz = $foo
  ? doFoo()
  : doBar();
```

**Especially** when assignment is involved.

But: I don't favor these "artistic changes" PRs when existing code is there and is fine. There's no real technical merit to it unless _everyone_ agrees on the readability. Which doesn't seem to be the case.

**Comment by taylorotwell** [comment]
I don't typically merge these types of PRs.

**Comment by GrahamCampbell** [comment]
The thing to remember is that the reverse transformation would also be a valid PR, and so these PRs are not desirable because we can remove them and add them back in for ever.

**Pull request closed without merging**